pkg/kubectl/sorter.go

```diff
@@ -316,3 +305,77 @@ func (r *RuntimeSort) OriginalPosition(ix int) int {
 	}
 	return r.origPosition[ix]
 }
+
+type TableSorter struct {
+	field string
+	obj   *metav1beta1.Table
+}
+
+func (t *TableSorter) Len() int {
+	return len(t.obj.Rows)
+}
+
+func (t *TableSorter) Swap(i, j int) {
+	t.obj.Rows[i], t.obj.Rows[j] = t.obj.Rows[j], t.obj.Rows[i]
+}
+
+func (t *TableSorter) Less(i, j int) bool {
+	iObj := t.obj.Rows[i].Object.Object
+	jObj := t.obj.Rows[j].Object.Object
+
+	var iValues [][]reflect.Value
+	var jValues [][]reflect.Value
+	var err error
+
+	parser := jsonpath.New("sorting").AllowMissingKeys(true)
+	err = parser.Parse(t.field)
+	if err != nil {
+		glog.Fatalf("sorting error: %v\n", err)
+	}
+
+	// TODO(juanvallejo): this is expensive for very large sets.
```

Comment:
Actually, we should built that array @smarterclayton mentioned not here, but rather in `SortObjects` method when we iterate through all objects to verify JSONPath values and then use that array for sorting. So I'd suggest moving this TODO up there.

Reply:
Nvmd, I only now realized we have two `RuntimeSorter` and `RuntimeSort`, sigh... We'll need to fix this up in 1.13.